File: Solutions/Solution509.java
class Solution {
    public int fib(int n) {
        if (n == 0 || n == 1) {
            return n;
        }

        int[] fibCalc = new int[n + 1];
        fibCalc[0] = 0;
        fibCalc[1] = 1;

        for (int i = 2; i <= n; i++) {
            fibCalc[i] = fibCalc[i - 2] + fibCalc[i - 1];
        }

        return fibCalc[n];
    }
}

File: Solutions/Solution509.rs
impl Solution {
    pub fn fib(n: i32) -> i32 {
        if n < 1 {
            return 0;
        } else if n == 1 {
            return 1;
        }

        let mut tail = 0;
        let mut head = 1;
        let mut sum = 0;

        for _ in 2..=n {
            sum = tail + head;
            tail = head;
            head = sum;
        }

        sum
    }
}
